extensions/thunder/strategies/thunder_fsdp.py

```diff
@@ -115,16 +160,32 @@ def setup_environment(self) -> None:
     def setup_module(self, module: Module) -> Module:
         import thunder
 
-        module = thunder.distributed.fsdp(
-            module,
-            device=self.root_device,
-            sharding_strategy=self.sharding_strategy,
-            bucketing_strategy=self.bucketing_strategy,
-            **self._fsdp_kwargs,
-        )
-
-        # NOTE @IvanYaschuck says that `fsdp(jit(model))` could be supported in the future so that the user owns the `jit` call.
-        # we would still `jit(fsdp(undo_jit(jit(model))))` internally
+        if (cd := thunder.compile_data(module)) is not None:
+            # the module was already jitted
+            if thunder.compile_stats(module).last_traces is not None:
+                raise RuntimeError(
+                    "You already called `thunder.jit()` and generated an execution trace. It's too late to apply the"
+                    " FSDP transform."
```

Comment:
That means the user has to remove their `thunder.jit()` call from the code. Can we recommend that in the error message?

Reply:
Not quite, what it means is that the user did something like

```python
model = thunder.jit(model)
model(x)
model = fabric.setup(model)
```

So what would need to be removed is the forward call. Does this look better?

```suggestion
                    "You already called `thunder.jit()` and generated an execution trace by running forward. It's too late to apply the"
                    " FSDP transform."
```

Reply:
Sure, what I mean is that generally we need to propose an action explicitly as in "do X to avoid error".